ote/src/clj/ote/tasks/pre_notices.clj

```diff
@@ -57,24 +59,37 @@
        (for [cell row]
          [:td cell])])]])
 
-(defn detected-change-row [{:keys [service-name operator-name change-date days-until-change
-                                   added-routes removed-routes changed-routes no-traffic-routes regions
-                                   date transport-service-id]}]
-  [operator-name
-   (str "<a href=\"" (environment/base-url) "#/transit-visualization/"
-        transport-service-id "/" date "\">" (escape-html service-name) "</a>")
-   (str/join ", " (db-util/PgArray->vec regions))
-   (str days-until-change " (" change-date ")")
-   (str/join ", "
-             (remove nil?
-                     [(when (and added-routes (> added-routes 0))
-                        (str added-routes " uutta reittiä"))
-                      (when (and removed-routes (> removed-routes 0))
-                        (str removed-routes " päättyvää reittiä"))
-                      (when (and changed-routes (> changed-routes 0))
-                        (str changed-routes " muuttunutta reittiä"))
-                      (when (and no-traffic-routes (> no-traffic-routes 0))
-                        (str no-traffic-routes " reitillä tauko liikennöinnissä"))]))])
+(defn detected-change-row [unsent-changes]
+  (let [grouped-changes (group-by :transport-service-id unsent-changes)]
+    (for [grouped-change grouped-changes]
+      (let [change-list (second grouped-change)
+            transport-service-id (:transport-service-id (first change-list))
+            operator-name (:operator-name (first change-list))
+            service-name (:service-name (first change-list))
+            date (:date (first change-list))
+            regions (:regions (first change-list))
+            days-until-change (:days-until-change (first change-list))
+            different-week-date (:different-week-date (first change-list))
+            added-routes (count (filter #(= "added" (:change-type %)) change-list))
+            removed-routes (count (filter #(= "removed" (:change-type %)) change-list))
+            changed-routes (count (filter #(= "changed" (:change-type %)) change-list))
+            no-traffic-routes (count (filter #(= "no-traffic" (:change-type %)) change-list))]
+
+        [operator-name
+         (str "<a href=\"" (environment/base-url) "#/transit-visualization/"
+              transport-service-id "/" date "\">" (escape-html service-name) "</a>")
+         (str/join ", " (db-util/PgArray->vec regions))
+         (str days-until-change " (" different-week-date ")")
+         (str/join ", "
+                   (remove nil?
+                           [(when (and added-routes (> added-routes 0))
```

Comment:
are these "and"s really necessary as added-routes is a number returned by count? Could also use pos? instead of (> added-routes 0).